collection_test.go

```diff
@@ -728,3 +691,74 @@ func ExampleCollectionSpec_LoadAndAssign() {
 	// Output: SocketFilter
 	// Array
 }
+
+var collectionSpecCmpOptions = cmp.Options{
+	cmp.Comparer(func(a, b asm.Metadata) bool {
+		return a == b
+	}),
+	cmp.Comparer(func(a, b *Program) bool {
+		return a == b
+	}),
+	cmp.Comparer(func(a, b *btf.Spec) bool {
+		// No way to compare a btf.Spec at the moment, this is enough for now.
+		return a == b
+	}),
+	// Empty MapSpec.Extra are equal.
+	cmp.Comparer(func(a, b bytes.Reader) bool {
+		if a.Len() == 0 && b.Len() == 0 {
+			return true
+		}
+		return false
+	}),
+}
+
+var collectionSpecIsEqual = qt.CmpEquals(collectionSpecCmpOptions...)
+
+func mustNewCollection(t *testing.T, spec *CollectionSpec, opts *CollectionOptions) *Collection {
```

Comment:
We need the error at least for TestLibBPFCompat, since that has custom logic around the error return. I could remove `testing.T` from `newCollection` and manually do cleanup in the callers if you prefer that.